libpkg/fetch_file.c

```diff
@@ -54,14 +54,25 @@ file_open(struct pkg_repo *repo, struct fetch_item *fi)
 		return (EPKG_FATAL);
 	}
 	u+=2;
-	/* if we don't have a '/' it means we have a host we should ignore */
+	/* if we don't have a '/' it means we have a host FQDN component, otherwise just proceed */
+	/* we can fetch local files only, so we accept the localhost FQDN */
+	/* TODO: consider accepting gethostname/getdomainname and combinations of these. */
+	/* TODO: delegate to curl to fetch any URL, btw. curl bails on this as well. */
 	if (*u != '/') {
-		u = strchr(u+1, '/');
-		if (u == NULL) {
-			pkg_emit_error("Invalid url: %s'\n', "
-					"file://<absolutepath> expected", fi->url);
+		char fqdn[256]="";
+		char *path = strchr(u+1, '/');
+		if (path == NULL) {
+			pkg_emit_error("Invalid url: '%s',\n"
+					"file:///<path> or file://localhost/<path> expected.", fi->url);
 			return (EPKG_FATAL);
 		}
+		strncat(fqdn, u, MIN(255, path-u));
```

Comment:
would be simpler to use xstrndup ?

Reply:
Yes, was considered, but if I know the maximum length of an URL (253) I prefer not to malloc/free.
But happy to change.

Reply:
URL^h^h^FQDN